packages/ember-runtime/lib/mixins/enumerable.js

```diff
@@ -49,8 +49,8 @@ function iter(key, value) {
   let valueProvided = arguments.length === 2;
 
   return valueProvided ?
-    (item)=> value === get(item, key) :
-    (item)=> !!get(item, key);
+    (item)=> value === item[key] :
```

Comment:
Hmm, this breaks `item[key]` being a computed property doesn't it?

Reply:
@rwjblue 

> being a computed property doesn't it?

considering the array, a computed property will not affect the behavior. 

``` javascript
/* normal array */
arr.findBy('u.s.a.'); // returns undefined. Same applicable for filterBy, isEvery and isAny. 

/* `arr` a computed property */
this.get(arr).findBy('u.s.a.'); // returns undefined
```
correct me, if i misunderstood the use case.

> Ember.get(arr, 'u.s.a'); // returns `undefined`

This was a falsy example mentioned in the description.

Reply:
Sorry, I’m saying that the property we are finding by is a computed property in the items. Using Ember.get today supports that (but is also the thing that allows nested path lookup), and I believe this change would break those use cases. 


For example:

```
let a = [
  { id: 1, foo: { bar: true } },
  { id: 2, foo: { bar: false } },
  { id: 3, foo: { bar: true } },
];

a.findBy(‘foo.bar’);
```

Reply:
@rwjblue  Thanks for pointing out. You were right. 👍 

I have handled that use case. Please check.
Handled for both `enumerable` and  `get prop`.